modules/codegen/src/smithy4s/codegen/IR.scala

```diff
@@ -114,15 +118,31 @@ object Field {
 
 }
 
-case class Alt(name: String, realName: String, tpe: Type, hints: List[Hint])
+sealed trait UnionMember {
+  def update(f: Product => Product)(g: Type => Type): UnionMember = this match {
+    case TypeCase(tpe)        => TypeCase(g(tpe))
+    case ProductCase(product) => ProductCase(f(product))
+  }
+}
+object UnionMember {
+  case class ProductCase(product: Product) extends UnionMember
+  case class TypeCase(tpe: Type) extends UnionMember
+}
+
+case class Alt(
```

Comment:
I agree, I didn't make it final since none of the other case classes in this file are final.. I felt awkward sticking out like that 😆  Happy to go either way on this

Reply:
@lewisjkl feel free to address in a later PR and apply everywhere, it's not a piece of code that's meant to be used as library really.